packages/nextjs/app/builders/[address]/page.tsx

```diff
@@ -2,11 +2,59 @@ import { notFound } from "next/navigation";
 import { UpgradedToBGCard } from "./_components/UpgradedToBGCard";
 import { UserChallengesTable } from "./_components/UserChallengesTable";
 import { UserProfileCard } from "./_components/UserProfileCard";
+import { Metadata } from "next";
+import { isAddress } from "viem";
 import { RouteRefresher } from "~~/components/RouteRefresher";
 import { isBgMember } from "~~/services/api-bg/builders";
 import { getLatestSubmissionPerChallengeByUser } from "~~/services/database/repositories/userChallenges";
 import { getUserByAddress } from "~~/services/database/repositories/users";
 
+type Props = {
+  params: {
+    address: string;
+  };
+};
+
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  const address = params.address;
+  const isValidAddress = isAddress(address);
+
+  // Default title and description
+  const title = "Builder Profile | Speed Run Ethereum";
```

Comment:
Could the title be dynamic too? `ens or short address | Speed Run Ethereum`

maybe we could return this data from the og endpoint too (in case you don't want to compute twice)

e.g. `const [addressOrEns, ogImageUrl] = /api..`

Reply:
I was able to make the title dynamic, I split out a new function for getting the ENS or Short Address at `packages/nextjs/utils/ens-or-address.ts`

I did try to have the og endpoint return both, but it resulted in the image not working.

Reply:
Ohh yeah :( dind't find the straigt forward way to make both working